enhancements/installer/manifests-from-stdin.md

```diff
@@ -0,0 +1,114 @@
+---
+title: manifests-from-sdtin
+authors:
+  - "@oglok"
+  - "@copejon"
+reviewers:
+  - "@fzdarsky"
+approvers:
+  - "@crawford"
+creation-date: 2020-12-24
+last-updated: 2020-12-24
+status: provisional
+---
+
+# Manifests from STDIN
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In many occasions, users need to customize the install-config target and/or the manifests generated by the OpenShift Installer in a reproducible manner. The most popular tool to allow this type of customization is “Kustomize”, and it is already well-known by the Kubernetes ecosystem. The result of any “Kustomization” is sent to standard output as a stream of manifests.
+
+This OpenShift enhancement proposes to allow the installer to read a stream of manifests from STDIN on the main targets that consume manifests.
+
+
+## Motivation
+
+Infrastructure-as-Code (e.g. GitOps) is an operational best practice for provisioning and managing large numbers of clusters and services. A typical IaC workflow involves pulling Kubernetes resource manifest templates of a specific version from a `git` repo, rendering those templates for a given cluster instance, and then running `openshift-install` (resp. `oc apply`) on the rendered manifests to install (resp. update) a cluster.
```

Comment:
I don't claim to be a GitOps aficionado, so perhaps this is a naive question, but what precludes leaving this entirely outside of the purview of the Installer? Would you not want to source control the actual rendered assets that were used rather than the template and mutations? What if the actual manifests created by `openshift create manifests` change over time given the same install-config.yaml input? Is that desirable or undesirable regardless of whether or not it breaks your pipeline.

Reply:
Both approaches (version controlling the one-off generated manifests vs the install-config.yaml + mutations) are valid and used, each with pros and cons.
In both cases, users would want to pin versions of all installation artefacts (`openshift-install`, `kustomize`, ...) matching the to-be-generated/consumed manifests, such that re-deploying the same git tag (e.g. for disaster recovery) at a later point in time yields the exact same result. In both cases, changes to newer installer versions may require updates to manifests.
One advantage of the "install-config + mutations" approach with `kustomize` is that it minimizes the delta that needs to be (version) controlled. So even when manifests created by the `create manifests` target drift (it's not a stable API after all), the "patches" are more likely to apply cleanly still.